pkg/rtc/wrappedreceiver.go

```diff
@@ -113,33 +113,21 @@ func (r *WrappedReceiver) DetermineReceiver(codec webrtc.RTPCodecCapability) boo
 		}
 	}
 	if trackReceiver == nil {
-		r.params.Logger.Errorw("can't determine receiver for codec", nil, "codec", codec.MimeType)
-		if len(r.receivers) > 0 {
-			trackReceiver = r.receivers[0]
-		}
+		r.lock.Unlock()
+		r.params.Logger.Warnw("can't determine receiver for codec", nil, "codec", codec.MimeType)
+		return false
 	}
 	r.TrackReceiver = trackReceiver
 
-	var onReadyCallbacks []func()
-	if trackReceiver != nil {
-		onReadyCallbacks = r.onReadyCallbacks
-		r.onReadyCallbacks = nil
-	}
+	onReadyCallbacks := r.onReadyCallbacks
+	r.onReadyCallbacks = nil
 	r.lock.Unlock()
 
-	if trackReceiver != nil {
-		for _, f := range onReadyCallbacks {
-			trackReceiver.AddOnReady(f)
-		}
-
-		if s, ok := trackReceiver.(*simulcastReceiver); ok {
-			if d, ok := s.TrackReceiver.(*DummyReceiver); ok {
-				return d.IsReady()
-			}
-		}
-		return true
+	for _, f := range onReadyCallbacks {
+		trackReceiver.AddOnReady(f)
 	}
-	return false
+
+	return true
```

Comment:
Not checking for ready status here as it is not really needed.

Checking for ready also showed a failure more. When across relay, if the subscriber is on the remote node, `DetermineReceiver` says ready because relay up track receiver is ready. That meant the subscribed quality change/subscribed audio codec change did not fire at all. It worked okay for video because there are more layer update messages happening and that triggers the same callback and that forced the codec regression to happen. Failed in audio case as this happens only once.

There is no need to check for ready because `DetermineReceiver` will queue up the ready callbacks on the `DummyReceiver` and when that gets upgraded to the real receiver, the on ready callbacks would fire and go from there.